### PR TITLE
Issue 667 : Extend environment data to be read from local file

### LINKED ISSUE #667: post_deploy hook not correctly retrieved
When you check the configuration with `vendor/bin/ece-tools wizard:ideal-state` the output is always `Your application does not have the "post_deploy" hook enabled.` In fact the hook is correctly configured in .magento.app.yaml. But the check in src/Config/Application/HookChecker.php reads the setting from a environment variable (ENV_APPLICATION) rather than from .magento.app.yaml. Instead Magento\MagentoCloud\Filesystem\FileList::getAppConfig() should get called to retrieve the setting from the file.

Is my assumption correct?

### PR DESCRIPTION
**Pull request by BarnyShergold**
### Description
Updated getApplication to try and read physical .magento.app.yaml if ENV is empty - this happens on local instances.
This will allow higher level uses to work on local installations. Specifically this was done to allow `vendor/bin/ece-tools wizard:ideal-state` to report correctly

### Fixed Issues (if relevant)

1. magento/ece-tools#667: post_deploy hook not correctly retrieved

### Manual testing scenarios

1. on a local instance run 'vendor/bin/ece-tools wizard:ideal-state' - if .magento.app.yaml is present and the "post_deploy" hook correctly configured then the return will be that the application is in an ideal state
2. Also running locally and REMOVING the file or having post_deploy hook not configured will cause normal warning message. Missing file will not cause an  untrapped exception.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/EnvironmentData.php
Comment (oshmyheliuk) [inline src/Config/EnvironmentData.php]: I would recommend removing comments as it's clear what code is doing.

File: src/Config/EnvironmentData.php
Comment (oshmyheliuk) [inline src/Config/EnvironmentData.php:147]: Instead of a comment, I think better to add some warning message about neither env variable and .magento.app.yaml  not exists

File: src/Config/EnvironmentData.php
Comment (oshmyheliuk) [inline src/Config/EnvironmentData.php]: Please add new line

File: src/Config/EnvironmentData.php
Comment (oshmyheliuk) [reply src/Config/EnvironmentData.php:147]: But I'm not sure about the message, @YPyltiai what do you think?

Comment (BarnyShergold) [reply src/Config/EnvironmentData.php:147]: Well, I left it as nothing as that is the current behaviour. I did think about putting a message in but left it. If the consensus is that at message is needed, should it be output direct to console line or done as a log? I'd lean towards the former as in theory it should only happen on a local instance if the file is missing.

Comment (oshmyheliuk) [reply src/Config/EnvironmentData.php:147]: I've discussed with @YPyltiai and we think that message is not needed. Let's leave as it is, thank you.

Comment (BarnyShergold) [reply src/Config/EnvironmentData.php:147]: Okay - I made the other changes you requested so can you approve the PR?
BTW I don't understand why the CI Checks are failing. The failure doesn't seem appropriate ?
@oshmyheliuk @YPyltiai

**Comment by BarnyShergold** [comment]
@bbatsche @oshmyheliuk @YPyltiai  - can we not get approval on this PR please? It had minor corrections made 4 days ago and we could get this into QA and moved on? Please? :)

File: src/Config/EnvironmentData.php
Comment (oshmyheliuk) [reply src/Config/EnvironmentData.php:147]: Yes, sorry, I thought that I've already approved

**Review by oshmyheliuk — DISMISSED** [review]

**Comment by shiftedreality** [comment]
Hello @BarnyShergold,

This issue is fixed for the Cloud Docker installation. Could you provide another example of a local installation?

**Comment by BarnyShergold** [comment]
@shiftedreality - Yes - currently Vaimo developers use Vagrant. This is what I tested it against.

**Review by bbatsche — DISMISSED** [review]

**Comment by shiftedreality** [comment]
@BarnyShergold could you please fix CI tests?

**Comment by shiftedreality** [comment]
There is an issue with secrets for community PRs, I'll need to fix it first to unblock this PR

**Comment by BarnyShergold** [comment]
@bbatsche  @oshmyheliuk @shiftedreality - All CI Checks have now passed (after SR fixed the CI a couple of small code formatting issues arose, which I fixed) and Travis has passed

Can you please re-review and then we can this PR moved on :)

**Review by oshmyheliuk — APPROVED** [review]

**Review by bbatsche — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Comment by oshmyheliuk** [comment]
@BarnyShergold Thank you for your contribution.

**Pull request merged**
